### PR TITLE
Validate release prep

### PR DESCRIPTION
**Pull request by n3wscott**

### CODE REVIEW

File: hack/update-deps.sh [hunk 1]
```diff
@@ -28,29 +28,24 @@ cd ${ROOT_DIR}
...
   shift
 done
-readonly GO_GET
+readonly UPGRADE
+readonly VERSION
 
-if (( GO_GET )); then
-    go get -d ${FLOATING_DEPS[@]}
+if (( UPGRADE )); then
+  FLOATING_DEPS=( $(run_go_tool tableflip.dev/buoy buoy float ${ROOT_DIR}/go.mod --release ${VERSION}) )
```
Comment (mattmoor) [inline hack/update-deps.sh:46]: add the domain explicitly here and above

Comment (n3wscott) [reply hack/update-deps.sh]: good idea.

File: .github/workflows/knative-releasability.yaml [hunk 2]
```diff
@@ -40,15 +41,27 @@ jobs:
...
+        with:
+          path: ./src/knative.dev/${{ github.event.repository.name }}
 
       - name: Check
-        run: |
-          buoy check go.mod --release $RELEASE --verbose
+        run: buoy check go.mod --release ${RELEASE} --domain knative.dev --verbose
+
+    - name: Upgrade
+      run: ./hack/update-deps.sh --release ${RELEASE} --upgrade
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:55]: the indentation changes 🤔

File: .github/workflows/knative-releasability.yaml [hunk 3]
```diff
@@ -40,15 +41,27 @@ jobs:
...
+
+    - name: Verify
+      run: |
+        if [[ -z "$(git status --porcelain)" ]]; then
+            echo "${{ github.repository }} up to date."
+        else
+            echo "${{ github.repository }} is out of date."
+            exit 1
+        fi
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:64]: I think we can use Markus' trick: https://github.com/mattmoor/knobots-actions/issues/7 to predicate the slack notification and customize its message without this *failing*

File: .github/workflows/knative-releasability.yaml (hunk 2 above)
Comment (n3wscott) [reply .github/workflows/knative-releasability.yaml]: my bad

File: .github/workflows/knative-releasability.yaml (hunk 3 above)
Comment (n3wscott) [reply .github/workflows/knative-releasability.yaml]: done

File: .github/workflows/knative-releasability.yaml [hunk 4]
```diff
@@ -40,31 +42,43 @@ jobs:
...
-          GO111MODULE=on go get tableflip.dev/buoy@master
+        run: GO111MODULE=on go get tableflip.dev/buoy@master
 
       - name: Check out code
         uses: actions/checkout@v2
+        with:
+          path: ./src/knative.dev/${{ github.event.repository.name }}
 
       - name: Check
+        run: buoy check go.mod --release ${RELEASE} --domain knative.dev --verbose
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:53]: We no longer notify when this fails 🙃 

I was thinking something along the lines of:
```shell
cat >> $GITHUB_ENV <<EOF
SLACK_MESSAGE(buoy check || CHECK=$?)
EOF
if [[ "$CHECK" -ne "0"  ]]; then
   echo 'notify=true' >> $GITHUB_ENV
fi

...

if [[ -z "$(git status --porcelain)" ]]; then
  ...
else
   echo 'notify=true' >> $GITHUB_ENV
   echo 'SLACK_MESSAGE=something about the dependencies being out of date'  >> $GITHUB_ENV
fi
```

Comment (n3wscott) [reply .github/workflows/knative-releasability.yaml:53]: right arg. sorry

File: .github/workflows/knative-releasability.yaml [hunk 5]
```diff
@@ -40,35 +42,65 @@ jobs:
...
         uses: actions/checkout@v2
+        with:
+          path: ./src/knative.dev/${{ github.event.repository.name }}
 
       - name: Check
         run: |
-          buoy check go.mod --release $RELEASE --verbose
+          echo 'CHECK_MESSAGE<<EOF' >> $GITHUB_ENV
+          (buoy check go.mod --release ${RELEASE} --domain knative.dev --verbose || CHECK=$1) >> $GITHUB_ENV
+          echo 'EOF' >> $GITHUB_ENV
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:56]: Not sure if this is more or less confusing:
```suggestion
          cat >> $GITHUB_ENV <<END
          CHECK_MESSAGE<<EOF
          $(buoy check go.mod --release ${RELEASE} --domain knative.dev --verbose || CHECK=$?)
          EOF
          END
```

Either way `$1` is wrong 😉

File: .github/workflows/knative-releasability.yaml [hunk 6]
```diff
@@ -40,35 +42,65 @@ jobs:
...
+          if [[ "$CHECK" -eq "0"  ]]; then
+            echo 'current=true' >> $GITHUB_ENV
+          else
+            echo 'current=false' >> $GITHUB_ENV
+          fi
+
+      - name: Upgrade
+        if: env.current == 'true'
+        run: |
+          if ./hack/update-deps.sh --release ${RELEASE} --upgrade; then
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:67]: What is this doing?

File: .github/workflows/knative-releasability.yaml (hunk 5 above)
Comment (n3wscott) [reply .github/workflows/knative-releasability.yaml:56]: my bash-fu is not good enough to understand things that use `>> sadsa <<`

My bad, I ment `$?`

File: .github/workflows/knative-releasability.yaml (hunk 6 above)
Comment (n3wscott) [reply .github/workflows/knative-releasability.yaml:67]: this is bash `if` testing the return code, so if the hack command returns 0 (success), bash does the `then`

Comment (n3wscott) [reply .github/workflows/knative-releasability.yaml]: Added some comments to help

File: .github/workflows/knative-releasability.yaml [hunk 7]
```diff
@@ -40,35 +42,68 @@ jobs:
...
 
       - name: Check out code
         uses: actions/checkout@v2
+        with:
+          path: ./src/knative.dev/${{ github.event.repository.name }}
 
       - name: Check
         run: |
-          buoy check go.mod --release $RELEASE --verbose
+          # The following pushes the sdtout of buoy into $CHECK_MESSAGE
```
Comment (mattmoor) [inline .github/workflows/knative-releasability.yaml:54]: stdout

**Review by mattmoor — COMMENTED** [review]
/lgtm

Let's see what it does.

**Pull request merged**